x-pack/legacy/plugins/transform/public/shared_imports.ts

```diff
@@ -37,3 +37,5 @@ export { setDependencyCache } from '../../ml/public/application/util/dependency_
 
 // @ts-ignore: could not find declaration file for module
 export { KqlFilterBar } from '../../ml/public/application/components/kql_filter_bar';
+
+export { useXJsonMode } from '../../../../plugins/watcher/public/application/sections/watch_edit/components/json_watch_edit/use_x_json_mode';
```

Comment:
i wonder if it’s worth making a copy of `use_x_json_mode`
deep including from other x-pack plugins isn’t recommended, plus watcher isn't long for this world as kibana alerts are on the horizon.

Reply:
I agree we should copy the file. The inner imports of that file are more obvious to reimport because they come from the `es_ui_shared` folder.

Reply:
Done in https://github.com/elastic/kibana/pull/58015/commits/031023a46b74b4a0333830b71cfe01d2c5e49450